### DISCUSSION TITLE
CSRF token to expire with session when config file exists

### DISCUSSION THREAD

**Discussion by NickiForte** [apache/airflow#36924] [General]
### Apache Airflow version

2.8.0

### What happened?

Despite https://github.com/apache/airflow/pull/28730, I encounter the message that the CSRF token has expired, although I checked that the session has not ended:

![image](https://github.com/apache/airflow/assets/95344621/836479cc-3259-490d-83f7-ded75e59fd76)

The reason why this still happens is because I have defined a custom webserver config, where the argument `WTF_CSRF_TIME_LIMIT` is not specified at all.

### What you think should happen instead?

As is discussed in https://github.com/apache/airflow/pull/28730, it is completely unnecessary for the CSRF token to expire before the end of a session, as it does not provide any extra security.

The default value of `WTF_CSRF_TIME_LIMIT` should be the same regardless of if there is a webserver config file or not, unless explicitly overridden.

### How to reproduce

Define a custom webserver config where the argument `WTF_CSRF_TIME_LIMIT` is not specified at all. Then just wait until the CSRF token expires.

And as is seen clearly in this function,
https://github.com/apache/airflow/blob/681859c7bffabce0c294060d811db2fb16851816/airflow/configuration.py#L2059C1-L2066C96,
if you have specified a custom webserver config, the values in the default webserver config are ignored. Thus the CSRF expiry time will take on the flask default, i.e. 3600 seconds, instead of the airflow default which is None.

### Operating System

MacOS

### Deployment

Amazon (AWS) MWAA

### Anything else?

It is of course possible to specify the argument `WTF_CSRF_TIME_LIMIT = None` in the custom webserver config. However, this CSRF expiry issue can be a real headache and it took quite some time to figure out the fix, so it would be nice to solve this once and for all.

One could also argue that if one creates a custom webserver config, one should copy the default values first. In my case, however, this was done before https://github.com/apache/airflow/pull/28730, so even if we now use a new airflow version, we still used the same config. I suspect this is the case for many others as well.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
I think it's reasonable to assume that default value from Flask is used if not specified. The problem is easy to fix on your side - just define `WTF_CSRF_TIME_LIMIT` and your problem is fixed. I see no reason to complicate the code and add our defaults for this particular case. If you define your custom config, you should know what defaults are. 

I don't believe anyone would want to spend any time on fixing it. But I leave it for discussion. If you @NickiForte   - or anyone - who would like to spend their time and work out the documentation, backwards-compatibility behaviour and implementation of it and make a comprehensive and complete PR i think we could consider reviewing and accepting it, but I do not see a reason to treat it as bug and keep open as issue.

Of course - if others have a different opinions - this is a discussion and opinions can still be stated here.
